### PR TITLE
Adding of a Municipal new tab in referent space

### PR DESCRIPTION
**Pull request by ktherage**

### CODE REVIEW

File: features/referent_space.feature [hunk 1]
```diff
@@ -18,3 +20,43 @@ Feature:
...
+    And I accept terms of use
+    Then I wait 2 seconds
+    And I should see "Banner"
+    And I should see "Bruce"
+    And I should see "+33 6 06 06 06 06"
+    And I should see "New York"
+    And I should see "Télécharger le CV"
+    And I should see "Oui"
+
+  @javascript @wip
```
Comment (ottaviano) [inline features/referent_space.feature]: to remove `@wip` 😄

File: src/Repository/RunningMateRequestRepository.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public function findForReferent(Adherent $referent): array
+    {
+        if (!$referent->isReferent()) {
+            return [];
+        }
+
+        return $this->createQueryBuilder('r')
+            ->select('r', 'CASE WHEN a.id IS NOT NULL THEN 1 ELSE 0 END AS isAdherent')
```
Comment (ottaviano) [inline src/Repository/RunningMateRequestRepository.php]: `createQueryBuilder('r')` already adds `from()` and `select('r')`
here you could just call `addSelect('CASE ...')`

File: features/referent_space.feature [hunk 3]
```diff
@@ -18,3 +20,43 @@ Feature:
...
+  Scenario: I can see running mate request for the zones i manage
+    Given I am logged as "referent-75-77@en-marche-dev.fr"
+    When I am on "/espace-referent/municipale/candidature-benevole"
+    And I wait 2 seconds
+    And I accept terms of use
+    Then I wait 2 seconds
+    And I should see "Stark"
+    And I should see "Tony"
+    And I should see ""
```
Comment (thomasbisignani) [inline features/referent_space.feature]: Oups ? :man_cartwheeling:

File: templates/referent/municipal/running_mate_request.html.twig [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    },
+                    {
+                        key: 'cvLink',
+                        name: 'Lien vers CV',
+                        link: true,
+                        targetBlank: true,
+                    },
+                    {
+                        key: 'isAdherent',
+                        name: 'Adherent (oui/non)',
```
Comment (ottaviano) [inline templates/referent/municipal/running_mate_request.html.twig]: `Adhérent` :)

File: templates/referent/municipal/volunteer_request.html.twig [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                        key: 'phone',
+                        name: 'Tél',
+                    },
+                    {
+                        key: 'favoriteCities',
+                        name: 'Ville(s) choisie(s)',
+                    },
+                    {
+                        key: 'isAdherent',
+                        name: 'Adherent (oui/non)',
```
Comment (ottaviano) [inline templates/referent/municipal/volunteer_request.html.twig]: Adhérent

File: src/Repository/RunningMateRequestRepository.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            ->leftJoin(Adherent::class, 'a', Join::WITH, 'a.emailAddress = r.emailAddress AND a.adherent = 1')
+            ->innerJoin('r.referentTags', 'tag')
+            ->andWhere('tag IN (:tags)')
+            ->setParameter('tags', $referent->getManagedArea()->getTags())
+            ->addOrderBy('r.firstName', 'ASC')
```
Comment (ottaviano) [inline src/Repository/RunningMateRequestRepository.php]: I think we should sort by lastName first

File: src/Repository/VolunteerRequestRepository.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            return [];
+        }
+
+        return $this->createQueryBuilder('r')
+            ->select('r', 'CASE WHEN a.id IS NOT NULL THEN 1 ELSE 0 END AS isAdherent')
+            ->leftJoin(Adherent::class, 'a', Join::WITH, 'a.emailAddress = r.emailAddress AND a.adherent = 1')
+            ->innerJoin('r.referentTags', 'tag')
+            ->andWhere('tag IN (:tags)')
+            ->setParameter('tags', $referent->getManagedArea()->getTags())
+            ->addOrderBy('r.firstName', 'ASC')
```
Comment (ottaviano) [inline src/Repository/VolunteerRequestRepository.php]: same here for the sort column

File: src/Repository/VolunteerRequestRepository.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function findForReferent(Adherent $referent): array
+    {
+        if (!$referent->isReferent()) {
+            return [];
+        }
+
+        return $this->createQueryBuilder('r')
+            ->select('r', 'CASE WHEN a.id IS NOT NULL THEN 1 ELSE 0 END AS isAdherent')
+            ->leftJoin(Adherent::class, 'a', Join::WITH, 'a.emailAddress = r.emailAddress AND a.adherent = 1')
```
Comment (ottaviano) [inline src/Repository/VolunteerRequestRepository.php]: could you add the same check `AND a.adherent = 1` in `IsAdherentAdminExtension` please?

File: features/bootstrap/FeatureContext.php [hunk 9]
```diff
@@ -95,4 +95,16 @@ public function clickElementSelector($cssElementSelector)
 
         $field->click();
     }
+
+    /**
+     * @Then I accept terms of use
+     */
+    public function acceptTermsOfUse()
+    {
+        $modalElement = 'body > div.ui-dialog.ui-widget.ui-widget-content.ui-corner-all.ui-front.ui-dialog-buttons.ui-draggable.ui-resizable > div.ui-dialog-buttonpane.ui-widget-content.ui-helper-clearfix > div > button';
```
Comment (thomasbisignani) [inline features/bootstrap/FeatureContext.php]: Is it possible to optimize the selector ?

File: features/referent_space.feature [hunk 10]
```diff
@@ -18,3 +20,43 @@ Feature:
     And I should see "En Marche - Projet citoyen"
     And I should see "Massive Open Online Course"
     And I should see "Formation en ligne ouverte à tous"
+
+  # Municipales space
+  @javascript
+  Scenario: I cannot see running mate request for the zones i don't manage
```
Comment (thomasbisignani) [inline features/referent_space.feature]: You should write the `i` in uppercase

File: templates/referent/municipal/volunteer_request.html.twig (hunk 5 above)
Comment (ktherage) [reply templates/referent/municipal/volunteer_request.html.twig]: Yeah, i'm angry with the accents :sweat_smile:

File: src/Repository/RunningMateRequestRepository.php [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+class RunningMateRequestRepository extends ServiceEntityRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, RunningMateRequest::class);
+    }
+
+    public function findForReferent(Adherent $referent): array
```
Comment (ottaviano) [inline src/Repository/RunningMateRequestRepository.php]: what about return just an array of the object (without `is_adherent` level)?
```php
$data = [];
foreach ($result as $row) {
  $row[0]->setAdherent($row['is_adherent']);
  $data[] = $row[0];
}
return $data;
```

File: src/Repository/VolunteerRequestRepository.php (hunk 7 above)
Comment (ottaviano) [reply src/Repository/VolunteerRequestRepository.php]: I unresolve this conversation :)

File: src/Repository/RunningMateRequestRepository.php (hunk 11 above)
Comment (ktherage) [reply src/Repository/RunningMateRequestRepository.php]: @ottaviano this column is needed to be displayed in the list.

**Pull request merged**